### PR TITLE
Implement game joining functionality

### PR DESCRIPTION
**Pull request by rjdellecese**
@lasley I moved the error logic into the model by raising there and rescuing in the model, and added tests for both model and controller. I'd appreciate any feedback, but especially regarding the way I handle checking for errors by a specific method in the model. Is there a better or different standard way to do that than I did?

### CODE REVIEW

File: app/controllers/games_controller.rb [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
   def show
     @game = Game.find(params[:id])
   end
+
+  def create
+    @game = current_user.games_as_white.create(game_params.merge(white_player_id: current_user))
+    @game.exception_if_name_too_short
+    redirect_to game_path(@game)
+  rescue => e
+    flash[:alert] = e.message
```
Comment (lasley) [inline app/controllers/games_controller.rb]: Perfect :+1:

File: app/controllers/games_controller.rb [hunk 2]
```diff
@@ -1,10 +1,41 @@
...
+    @game = Game.new
   end
 
   def show
     @game = Game.find(params[:id])
   end
+
+  def create
+    @game = current_user.games_as_white.create(game_params.merge(white_player_id: current_user))
+    @game.exception_if_name_too_short
```
Comment (lasley) [inline app/controllers/games_controller.rb]: Why not just call this in the create method? That way we guarantee that the exception is generated no matter the controller we use :wink:

Comment (lasley) [inline app/controllers/games_controller.rb]: More Rails-esque may actually be to use a validation instead

**Comment by lasley** [comment]
@rjdellecese Looks great, just one comment. You covered all the edge cases I can think of on the tests too :+1:

**Comment by lasley** [comment]
Re whether to use a method. General rule - take a step back, will you ever use this logic in another context? If yes, make a method. If no, either abstract the logic to make it reusable or just put it inline

**Comment by rjdellecese** [comment]
@lasley I think this should be ready to go now!
